File: web/src/api/campusAuth.js
import service from '@/utils/request'

export const getCampusAuthList = (params) => {
  return service({
    url: '/campusAuth/getCampusAuthList',
    method: 'get',
    params
  })
}

export const findCampusAuth = (params) => {
  return service({
    url: '/campusAuth/findCampusAuth',
    method: 'get',
    params
  })
}

export const submitCampusAuthTest = (data) => {
  return service({
    url: '/campusAgentReviewTest/submit',
    method: 'post',
    data
  })
}

export const reviewCampusAuth = (data) => {
  return service({
    url: '/campusAuth/reviewCampusAuth',
    method: 'post',
    data
  })
}

export const rejectCampusAuth = (data) => {
  return service({
    url: '/campusAuth/rejectCampusAuth',
    method: 'post',
    data
  })
}

export const revokeCampusAuth = (data) => {
  return service({
    url: '/campusAuth/revokeCampusAuth',
    method: 'post',
    data
  })
}

File: server/router/campus/campus_auth.go
package campus

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type CampusAuthRouter struct{}

func (r *CampusAuthRouter) InitCampusAuthRouter(privateRouter *gin.RouterGroup, publicRouter *gin.RouterGroup) {
	campusAuthRouter := privateRouter.Group("campusAuth").Use(middleware.OperationRecord())
	campusAuthRouterWithoutRecord := privateRouter.Group("campusAuth")
	campusAuthTestRouter := privateRouter.Group("campusAgentReviewTest").Use(middleware.OperationRecord())
	agentReviewPublicRouter := publicRouter.Group("api/agent/review")
	{
		campusAuthRouter.POST("reviewCampusAuth", campusAuthApi.ReviewCampusAuth)
		campusAuthRouter.POST("rejectCampusAuth", campusAuthApi.RejectCampusAuth)
		campusAuthRouter.POST("revokeCampusAuth", campusAuthApi.RevokeCampusAuth)
	}
	{
		campusAuthRouterWithoutRecord.GET("getCampusAuthList", campusAuthApi.GetCampusAuthList)
		campusAuthRouterWithoutRecord.GET("findCampusAuth", campusAuthApi.FindCampusAuth)
	}
	{
		campusAuthTestRouter.POST("submit", campusAuthApi.SubmitCampusAuthTest)
		agentReviewPublicRouter.POST("callback", campusAuthApi.AgentReviewCallback)
	}
}
